flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/MethodCallGen.scala

```diff
@@ -34,13 +33,14 @@ class MethodCallGen(method: Method) extends CallGenerator {
       returnType: LogicalType): GeneratedExpression = {
     generateCallIfArgsNotNull(ctx, returnType, operands, !method.getReturnType.isPrimitive) {
       originalTerms => {
-        val terms = originalTerms.zip(method.getParameterTypes).map { case (term, clazz) =>
-          // convert the BinaryString parameter to String if the method parameter accept String
-          if (clazz == classOf[String]) {
-            s"$term.toString()"
-          } else {
-            term
-          }
+        val terms = originalTerms.zipWithIndex.zip(method.getParameterTypes).map {
```

Comment:
What do you want to change?

Reply:
incorrect intermediate state. the zipWithIndex is introduced to fix the original TO_TIMESTAMP. I will remove it.